lib/info-extras.js

```diff
@@ -124,11 +124,32 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ','\n');
   try {
-    jsonStr = JSON.parse(jsonStr);
+    jsonStr = jsonStr.substring(0,jsonStr.length-1);
+    jsonStr = JSON.parse(jsonStr)
+    watch_next_json = JSON.parse(jsonStr["watch_next_response"])
+    secondary_results = watch_next_json.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results
+    recomendedVideos = [];
+    for(i in secondary_results)
+    {
+      let videoDetails = {};
+      if(secondary_results[i].compactVideoRenderer !== undefined)
+      {
+        let result = secondary_results[i].compactVideoRenderer
+        videoDetails["id"] = result.videoId
+        videoDetails["title"] = result.title.simpleText
+        videoDetails["author"] = result.shortBylineText.runs[0].text
+        videoDetails["ucid"] = result.shortBylineText.runs[0].navigationEndpoint.browseEndpoint.browseId
+        videoDetails["author_thumbnail"] = result.channelThumbnail.thumbnails[0].url
+        videoDetails["short_view_count_text"] = result.shortViewCountText.simpleText
+        videoDetails["view_count"] = result.viewCountText.simpleText.replace(" views","") || 0
+        videoDetails["length_seconds"] = result.lengthText.simpleText || "0.00"
+        recomendedVideos.push(videoDetails)
+      }
+    }
+    return recomendedVideos
   } catch (err) {
     return [];
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
```

Comment:
move everything that won't throw out of the `try` block, only keeping `JSON.parse` there. if the reason for everything else being in there is because some of those variables might later be undefined, then put another `try` block inside the for loop so that when going through each item, one being broken won't mean returning an empty list